src/core/Flyout.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from "react";
+import {
+  NavigationMenu,
+  NavigationMenuItem,
+  NavigationMenuList,
+  NavigationMenuTrigger,
+  NavigationMenuContent,
+  NavigationMenuViewport,
+  NavigationMenuLink,
+} from "@radix-ui/react-navigation-menu";
+import cn from "./utils/cn";
+
+const NavigationMenuDemo = ({
+  menuItems,
+  navMenuStyling,
+  flyOutStyling,
+}: {
+  menuItems: {
+    label: string;
+    content?: React.ReactNode;
+    link?: string;
+    panelStyling?: string;
+  }[];
+  navMenuStyling?: string;
+  flyOutStyling?: string;
+}) => {
+  const menuLinkStyles =
+    "ui-text-menu3 text-neutral-1000 font-bold px-12 py-8 hover:bg-neutral-100 hover:text-neutral-1300 cursor-pointer flex items-center";
+  const viewPortStyling =
+    "relative overflow-hidden w-full h-[var(--radix-navigation-menu-viewport-height)] origin-[top_center] transition-[width,_height] duration-300 data-[state=closed]:animate-scaleOut data-[state=open]:animate-scaleIn sm:w-[var(--radix-navigation-menu-viewport-width)]";
+  const boxShadowViewport =
+    "shadow-[0_201px_56px_0_rgba(20,25,36,0),0_129px_51px_0_rgba(20,25,36,0.02),0_72px_43px_0_rgba(20,25,36,0.06),0_32px_32px_0_rgba(20,25,36,0.10),0_8px_18px_0_rgba(20,25,36,0.12)]";
+
+  return (
+    <NavigationMenu className={cn(navMenuStyling, "flex w-full relative")}>
+      <NavigationMenuList className="flex list-none center">
+        {menuItems.map((menuItem) => (
+          <NavigationMenuItem key={menuItem.label}>
+            {menuItem.content ? (
+              <>
+                <NavigationMenuTrigger
+                  className={cn(
+                    menuLinkStyles,
```

Comment:
It's better to put the prop-driven styles last when inside a `cn` helper, as it allows them to gracefully override anything static (your second set of classes, "group outline-none ..."). `cn` includes `tailwind-merge` which gives priority in terms of order. Same applies to other `cn` uses here.

The exception to this is if you _want_ your static classes to be the highest priority - I leave this up to you.